src/App.js

```diff
@@ -1,25 +1,60 @@
-import logo from './logo.svg';
-import './App.css';
-
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+import 'bootstrap/dist/css/bootstrap.min.css';
+import React from 'react';
+import axios from 'axios';
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      searchQuery: '',
+      location: {},
+      error: false,
+    }
+  }
+
+  getLocation = async () => {
+
+    const url = `https://us1.locationiq.com/v1/search.php?key=${process.env.REACT_APP_CITY_KEY}&q=${this.state.searchQuery}&format=json`;
+
+    try {
+
+      const response = await axios.get(url);
+
+      const location = response.data[0];
+
+      this.setState({
+        location, // or location:location
+        error: false,
+      });
+
+    } catch (error) {
+
+      console.error('Unable to find city', this.state.searchQuery);
+
+      this.setState({ error: true });
+    }
+
+
+  }
+
+  render() {
+    return (
+      <>
+        <input onChange={(event) => this.setState({ searchQuery: event.target.value })} placeholder="search for a city"></input>
+        <button onClick={this.getLocation}>Explore!</button>
```

Comment:
Feedback here: In class we talked briefly about why making calls to setState here is not recommended (can cause infinite loops, make components less maintainable etc. In general we strive to keep render pure 👍 )

Instead, create a method outside of render to handle the event. Grab a TA and see if they can lend a hand :)